File: src/java/magma/app/compile/value/ConstructionCaller.java
package magma.app.compile.value;

import magma.api.option.None;
import magma.api.option.Option;
import magma.app.io.Platform;

public record ConstructionCaller(String right, Platform platform) implements Caller {
    @Override
    public String generate() {
        if (Platform.Magma == this.platform) {
            return this.right;
        }
        return "new " + this.right;
    }

    @Override
    public Option<Value> findChild() {
        return new None<Value>();
    }
}

File: src/node/magma/app/compile/value/ConstructionCaller.ts
import { Caller } from "../../../../magma/app/compile/value/Caller";
import { Platform } from "../../../../magma/app/io/Platform";
import { Value } from "../../../../magma/app/compile/value/Value";
import { Option } from "../../../../magma/api/option/Option";
import { None } from "../../../../magma/api/option/None";
export class ConstructionCaller implements Caller {
	right: string;
	platform: Platform;
	constructor (right: string, platform: Platform) {
		this.right = right;
		this.platform = platform;
	}
	generate(platform: Platform): string {
		if (Platform/*auto*/.Magma === this/*auto*/.platform) {
			return this/*auto*/.right;
		}
		return "new " + this/*auto*/.right;
	}
	findChild(): Option<Value> {
		return new None<Value>(/*auto*/);
	}
}
